### ISSUE TITLE
Error exporting Theme

### ISSUE THREAD

**Issue by chriskzn** [WordPress/create-block-theme#534]
Hi, I am receiving this error when trying to use the simple "Export Theme" option or any of the other options available.

Fatal error: Uncaught Error: Call to undefined method WP_Error::addFromStringToTheme() in C:\xampp\htdocs\olivier_blocks\wp-content\plugins\create-block-theme\admin\create-theme\theme-zip.php:81 Stack trace: #0 C:\xampp\htdocs\olivier_blocks\wp-content\plugins\create-block-theme\admin\class-create-theme.php(309): Theme_Zip::copy_theme_to_zip(Object(WP_Error), NULL, NULL) #1 C:\xampp\htdocs\olivier_blocks\wp-content\plugins\create-block-theme\admin\class-create-theme.php(477): Create_Block_Theme_Admin->export_theme(Array) #2 C:\xampp\htdocs\olivier_blocks\wp-includes\class-wp-hook.php(324): Create_Block_Theme_Admin->blockbase_save_theme('') #3 C:\xampp\htdocs\olivier_blocks\wp-includes\class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array) #4 C:\xampp\htdocs\olivier_blocks\wp-includes\plugin.php(517): WP_Hook->do_action(Array) #5 C:\xampp\htdocs\olivier_blocks\wp-admin\admin.php(175): do_action('admin_init') #6 C:\xampp\htdocs\olivier_blocks\wp-admin\themes.php(10): require_once('C:\\xampp\\htdocs...') #7 {main} thrown in C:\xampp\htdocs\olivier_blocks\wp-content\plugins\create-block-theme\admin\create-theme\theme-zip.php on line 81
There has been a critical error on this website. Please check your site admin email inbox for instructions.

Learn more about troubleshooting WordPress.

**Comment by creativecoder** [comment]
It looks like this might be a duplicate of, or very similar to #492 .

In order to export a theme zip, the site needs to have access to the [Zip extension for PHP](https://www.php.net/manual/en/book.zip.php), to be able to create .zip files. My guess would be that your site doesn't currently have that extension enabled.

Maybe you can check and see if that PHP extension can be enabled in xampp?

**Comment by chriskzn** [comment]
Thank you, you are correct, I needed to activate the extension=zip in my  \xampp\php\php.ini file. I had to stop and start Apache services and it is working. I wonder if there is not a method that can be used to check if the current extension is enabled and that could warn the user that the 'zip extension' is not currently enabled? I think this would be good once the user clicks on the 'Create Block Theme' option?

if (extension_loaded('zip')) {
    echo 'The zip extension is enabled.';
} else {
    echo 'The zip extension is not enabled.';
}
